src/Random/NormalDistribution.h

```diff
@@ -0,0 +1,214 @@
+     * @return The PDF value.
+    */
+    template<typename T>
+    KOKKOS_FUNCTION T normal_pdf_func(T x, T mean, T stddev) {
+      const T pi = Kokkos::numbers::pi_v<T>;
+      return (1.0 / (stddev * Kokkos::sqrt(2 * pi))) * Kokkos::exp(-(x - mean) * (x - mean) / (2 * stddev * stddev));
+    }
+
+    /*!
+     * @brief An estimator for the initial guess that is used in Newton-Raphson method of Inverste Transfrom Sampling
+     *
+     * @param x Input value.
+     * @param mean Mean of the distribution.
+     * @param stddev Standard deviation of the distribution.
+     * @return The estimate value.
+    */
+    template<typename T>
+    KOKKOS_FUNCTION T normal_estimate_func(T u, T mean, T stddev) {
+      const T pi = Kokkos::numbers::pi_v<T>;
+      return (Kokkos::sqrt(pi / 2.0) * (2.0 * u - 1.0)) * stddev + mean;
+    }
+
+     /*!
+     * @struct normal_cdf
+     * @brief Functor to calculate the cumulative distribution function (CDF) for a normal distribution.
+     *
+     * This functor calculates the CDF for a normal distribution in a specific dimension 'd'.
+     *
+     * @tparam T Data type for the input value 'x'.
+    */
+    template <typename T>
+    struct normal_cdf{
+      KOKKOS_INLINE_FUNCTION double operator()(T x, unsigned int d, const T *params) const {
+              T mean = params[2*d + 0];
+              T stddev = params[2*d + 1];
+              return ippl::random::normal_cdf_func<T>(x, mean, stddev);
+      }
+    };
+
+    /*!
+     * @struct normal_pdf
+     * @brief Functor to calculate the probability density function (PDF) for a normal distribution.
+     *
+     * This functor calculates the PDF for a normal distribution in a specific dimension 'd'.
+     *
+     * @tparam T Data type for the input value 'x'.
+    */
+    template <typename T>
+    struct normal_pdf{
+      KOKKOS_INLINE_FUNCTION double operator()(T x, unsigned int d, T const *params) const {
+              T mean = params[2*d + 0];
+              T stddev = params[2*d + 1];
+              return ippl::random::normal_pdf_func<T>(x, mean, stddev);
+      }
+    };
+
+     /*!
+     * @struct normal_estimate
+     * @brief Functor to estimate the initial guess for sampling normal distribution.
+     *
+     * This functor estimates the value for a normal distribution in a specific dimension 'd'.
+     *
+     * @tparam T Data type for the input value 'u'.
+    */
+    template <typename T>
+    struct normal_estimate{
+      KOKKOS_INLINE_FUNCTION double operator()(T u, unsigned int d,  T const *params) const {
+              T mean = params[2*d + 0];
+              T stddev = params[2*d + 1];
+              return ippl::random::normal_estimate_func<T>(u, mean, stddev);
+      }
+    };
+
+    /*!
+       * @file NormalDistribution.h
+       * @class NormalDistribution
+    */
+    template<typename T, unsigned Dim>
+    class NormalDistribution : public ippl::random::Distribution<T, Dim, 2*Dim,
```

Comment:
It is a bit weird that you have a class in this file which is almost empty and then a whole lot of different structs. Why can't the functions be members of this class and why do you need a different struct randn when we have the NormalDistribution class? Is it to differentiate between global and local normal random number generations?

Reply:
randn generates samples of normal distribution on [-infty, infty]^Dim (global).

The distribution class is only used in the InverseTransfromSampling for sampling on bounded domain [a,b] (local).
This class needs pdf, cdf and estimate as template parameters.
Normal distribution on bounded domain [a,b] is an example (child) of the distribution class where again pdf, cdf, estimate needs to be given as template parameters.

I've put together all the things associated with normal distribution in the same file.

Reply:
After discussing with @matt-frey we came up with the following things regarding the structure
1. Please avoid using a lot of stand alone C-style functions and structs like `normal_estimate_func`, `normal_estimate` and incorporate them into the class of NormalDistribution (as methods). Please do the same if a similar thing exists in other parts of your code. We would like to avoid stand alone functions and structs as much as possible to fit into the hierarchical class structure of C++.
2. Separate the "local" and "global" (sampling on the whole domain which you refer as `randn`) into separate files as they are fundamentally different in nature. In general we want to have one class per file and the name of the file is same as that of the class (there are few exceptions for this in some parts of IPPL but please stick to it as much as possible) and not group things of similar nature into one file. 
3. Please incorporate the missing items in the initial implementation https://github.com/matt-frey/ippl/blob/pic-manager/src/Random/Random.h in the global distribution (like functions mean and std for normal distribution) and try to follow it closely as it is very similar to std random number generator functions.

Reply:
NormalDistribution class inherits from Distribution class. The pdf/cdf/estimate functors combined in one struct is needed as template parameters for the base class.

```
template<typename T, unsigned Dim>
    class NormalDistribution : public ippl::random::Distribution<T, Dim, 2*Dim, NormalDistributionFunctions<T>>{
```

If we put the definition of `struct NormalDistributionFunctions` inside `class NormalDistribution`, then we can not use it in the template argument of base class (Distribution) I think. If you know how it can be done, please let me know.